### PR TITLE
Took 318 keyframe animations override transform of entity

### PR DESCRIPTION
**Pull request by fcturan20**
https://oyunteknolojileri.atlassian.net/browse/TOOK-292?atlOrigin=eyJpIjoiOTE1NjRjYTdlYWIwNDRlODlhNjZiNDcwOTMzOGZhY2YiLCJwIjoiaiJ9
Fixed: Deleted prefabs was causing various crashes.
Fixed: Changing prefab entity’s parent in Outliner was causing various entity bugs.
Added: GetAnimRecord to AnimControllerComponent as @wotanfinrod requested
Added: Prefab icon in OutlinerWindow

### CODE REVIEW

File: Editor/Action.cpp [hunk 1]
```diff
@@ -4,6 +4,8 @@
 #include "EditorScene.h"
 #include "GlobalDef.h"
 
+#include <Prefab.h>
```
Comment (afraidofdark) [inline Editor/Action.cpp:7]: use <> only for system libraries.

File: ToolKit/ResourceComponent.cpp [hunk 2]
```diff
@@ -325,11 +335,16 @@ namespace ToolKit
...
+
+    map = nullptr;
   }
 
   SkeletonComponent::~SkeletonComponent()
   {
-    delete map;
+    if (map)
+    {
+      delete map;
```
Comment (afraidofdark) [inline ToolKit/ResourceComponent.cpp:346]: use safedel for single pointers sefdelarr for arrays

File: ToolKit/ResourceComponent.h [hunk 3]
```diff
@@ -155,6 +155,7 @@ namespace ToolKit
     void Stop();
     void Pause();
     AnimRecordPtr GetActiveRecord();
+    AnimRecordPtr GetAnimRecord(const String& signalName);
```
Comment (afraidofdark) [inline ToolKit/ResourceComponent.h:158]: will this be returning the frist record with the signal name ? There is no guarantee that signal name for being unique.

**Review by afraidofdark — APPROVED** [review]
small fixes anda caviate note

File: ToolKit/ResourceComponent.h (hunk 3 above)
Comment (fcturan20) [reply ToolKit/ResourceComponent.h:158]: Editor doesn't add a new record if a record with the same name exists, so probably this won't be an issue. But as the std::unordered_map works, first one found is returned.

**Pull request merged**